pyboreas/data/himmelsbach.py

```diff
@@ -153,17 +155,16 @@ def compute_model_and_get_inliers(self):
                         line_set = []
                         c += 1
                 else:
-                    if (len(lines) > 0):
-                        dprev = self.distpointline(lines[c - 1], idx)
-                    else:
-                        dprev = -1
-                    if (dprev <= self.Tdprev or c == 0 or len(line_set) != 0):
+                    mprev = np.abs(lines[-1].m) if len(lines) > 0 else 0
+                    dprev = self.distpointline(lines[c - 1], idx) if len(lines) > 0 else -1
+                    if (mprev > self.Tm or dprev <= self.Tdprev or c == 0 or len(line_set) != 0):
                         line_set.append(idx)
                     i += 1
-            if len(lines) == 0:
+            if len(line_set) >= 2:
```

Comment:
looks like a bug, this if block should check and fit a line to whatever left in the line_set